src/papi-client.ts

```diff
@@ -0,0 +1,101 @@
+import {EventMessage} from "./models";
+import {URL, UUID} from "./types";
+import * as Promise from "bluebird";
+import {get, post, put, Response, SuperAgentRequest} from "superagent";
+
+export type RequestBuilder = (url: string, callback?: (err: any, res: Response) => void) => SuperAgentRequest;
+export type BatchId = number;
+
+export interface PapiCredentials {
+    login: string;
+    password: string;
+}
+const DEFAUT_DELAY_MS = 1000;
+
+export class PapiClient {
+
+    public currentBatchId: BatchId;
+
+    constructor(
+        private apiUrl: URL,
+        private domainUUID: UUID,
+        private credentials: PapiCredentials
+    ) {}
+
+    public startBatch(): Promise<Response> {
+        if (this.currentBatchId) {
+            throw new Error("The following batch is already started: " + this.currentBatchId);
+        }
+
+        return Promise.fromCallback(callback => this.request(post, "/batches/").end(callback))
```

Comment:
I did a `promisify` method taking a "request" and returning a "promise".
I need to keep the `request` method returning a SuperAgentRequest to be able to customize it .. see the `importICS` or the `waitForBatchSuccess` methods.